mesonbuild/modules/features/feature.py

```diff
@@ -0,0 +1,356 @@
+
+@dataclass(unsafe_hash=True, order=True)
+class ConflictAttr:
+    """
+    Data class representing an feature attribute that may conflict
+    with other features attributes.
+
+    The reason behind this class to clear any possible conflicts with
+    compiler arguments when they joined together due gathering
+    the implied features or concatenate non-implied features.
+
+    Attributes:
+        val: The value of the feature attribute.
+        match: Regular expression pattern for matching conflicted values
+               (optional).
+        mfilter: Regular expression pattern for filtering these conflicted values
+               (optional).
+        mjoin: String used to join filtered values (optional)
+
+    """
+    val: str = field(hash=True, compare=True)
+    match: Union[re.Pattern, None] = field(
+        default=None, hash=False, compare=False
+    )
+    mfilter: Union[re.Pattern, None] = field(
+        default=None, hash=False, compare=False
+    )
+    mjoin: str = field(default='', hash=False, compare=False)
+
+    def copy(self) -> 'ConflictAttr':
+        return ConflictAttr(**self.__dict__)
+
+    def to_dict(self) -> Dict[str, str]:
+        ret: Dict[str, str] = {}
+        for attr in ('val', 'mjoin'):
+            ret[attr] = getattr(self, attr)
+        for attr in ('match', 'mfilter'):
+            val = getattr(self, attr)
+            if not val:
+                val = ''
+            else:
+                val = str(val)
+            ret[attr] = val
+        return ret
+
+class KwargConfilctAttr(KwargInfo):
+    def __init__(self, func_name: str, opt_name: str, default: Any = None):
+        types = (
+            NoneType, str, ContainerTypeInfo(dict, str),
+            ContainerTypeInfo(list, (dict, str))
+        )
+        super().__init__(
+            opt_name, types,
+            convertor = lambda values: self.convert(
+                func_name, opt_name, values
+            ),
+            default = default
+        )
+
+    @staticmethod
+    def convert(func_name:str, opt_name: str, values: 'IMPLIED_ATTR',
+                ) -> Union[None, List[ConflictAttr]]:
+        if values is None:
+            return None
+        ret: List[ConflictAttr] = []
+        values = [values] if isinstance(values, (str, dict)) else values
+        accepted_keys = ('val', 'match', 'mfilter', 'mjoin')
+        for edict in values:
+            if isinstance(edict, str):
+                if edict:
+                    ret.append(ConflictAttr(val=edict))
+                continue
+            if not isinstance(edict, dict):
+                # It shouldn't happen
+                # TODO: need exception here
+                continue
+            unknown_keys = [k for k in edict.keys() if k not in accepted_keys]
+            if unknown_keys:
+                raise MesonException(
+                    f'{func_name}: unknown keys {unknown_keys} in '
+                    f'option {opt_name}'
+                )
```

Comment:
My feeling is that for ~100 characters it is still acceptable to be on one line when dealing with effectively just very long strings. Failing that, I'd still prefer to see:

```python
msg = 'long message here'
raise MesonException(msg)
```